ansible/roles/kubernetes-node/tasks/main.yml

```diff
@@ -5,5 +5,5 @@
   ignore_errors: True
 
 - name: join nodes to masters
-  command: "/usr/bin/kubeadm join {{ kuberenetes_common.api_ip }}:6443 --token='{{ hostvars[groups['masters'][0]]['bootstrap_token'] }}'"
+  command: "/usr/bin/kubeadm join {{ kubernetes_common.api_ip }}:6443 --token='{{ hostvars[groups['primary_master'][0]]['generated_token']['stdout'] }}' --discovery-token-unsafe-skip-ca-verification"
```

Comment:
any particular reason `[0]` is used instead of `|first` here?

Reply:
`first` is a jinja filter, and wouldn't make sense here when trying to index into an object.